com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -1,124 +1,232 @@
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.commons.collections4.ListUtils;
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.CloudControllerException;
+import org.cloudfoundry.client.lib.CloudException;
+import org.cloudfoundry.client.lib.CloudOperationException;
+import org.cloudfoundry.client.lib.CloudServiceBrokerException;
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.cloudfoundry.client.lib.domain.CloudService;
+import org.cloudfoundry.client.lib.domain.CloudServiceBinding;
+import org.cloudfoundry.client.lib.domain.CloudServiceInstance;
+import org.cloudfoundry.client.lib.domain.CloudServiceKey;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+import org.springframework.http.HttpStatus;
+
+import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.client.lib.domain.ImmutableCloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.core.model.ServiceOperation;
+import com.sap.cloud.lm.sl.cf.core.security.serialization.SecureSerializationFacade;
+import com.sap.cloud.lm.sl.cf.process.Constants;
+import com.sap.cloud.lm.sl.cf.process.Messages;
+import com.sap.cloud.lm.sl.cf.process.util.ExceptionMessageTailMapper;
+import com.sap.cloud.lm.sl.cf.process.util.ExceptionMessageTailMapper.CloudComponents;
+import com.sap.cloud.lm.sl.cf.process.util.ServiceOperationGetter;
+import com.sap.cloud.lm.sl.cf.process.util.ServiceProgressReporter;
+import com.sap.cloud.lm.sl.common.SLException;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+
+@Named("deleteServicesStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class DeleteServicesStep extends AsyncFlowableStep {
+
+    private final SecureSerializationFacade secureSerializer = new SecureSerializationFacade();
+
+    private ServiceOperationGetter serviceOperationGetter;
+    private ServiceProgressReporter serviceProgressReporter;
+
+    @Inject
+    public DeleteServicesStep(ServiceOperationGetter serviceOperationGetter, ServiceProgressReporter serviceProgressReporter) {
+        this.serviceOperationGetter = serviceOperationGetter;
+        this.serviceProgressReporter = serviceProgressReporter;
+    }
+
+    @Override
+    protected StepPhase executeAsyncStep(ExecutionWrapper execution) {
+        getStepLogger().debug(Messages.DELETING_SERVICES);
+
+        CloudControllerClient client = execution.getControllerClient();
+
+        List<String> servicesToDelete = new ArrayList<>(StepsUtil.getServicesToDelete(execution.getContext()));
+
+        if (servicesToDelete.isEmpty()) {
+            getStepLogger().debug(Messages.MISSING_SERVICES_TO_DELETE);
+            return StepPhase.DONE;
+        }
+
+        List<CloudServiceExtended> servicesData = getServicesData(servicesToDelete, execution);
```

Comment:
As mentioned earlier - I do not understand the suffix 'Data' here. The method just retrieves Service Instances so 'getServices' seems sufficient. What purpose does the 'Data' suffix serve here?
This includes variables and methods. Below is another variable and method containing the 'Data' suffix again.

Reply:
This class was returned as it is due backward compatability. This class will be removed after the release of these changes. I can rename this method and variable and add comment at the begging that this class will be removed.

Reply:
Perhaps put the @Deprecated annotation as well and explain why it will be removed